pom.xml

```diff
@@ -898,8 +898,10 @@
           <instructions>
             <!-- stops the "uses" clauses being added to "Export-Package" manifest entry -->
             <_nouses>true</_nouses>
+            <!-- remove extra headers to improve bit for bit reproducibility -->
+            <_noextraheaders>true</_noextraheaders>
             <!-- Stop the JAVA_1_n_HOME variables from being treated as headers by Bnd -->
-            <_removeheaders>JAVA_1_3_HOME,JAVA_1_4_HOME,JAVA_1_5_HOME,JAVA_1_6_HOME,JAVA_1_7_HOME,JAVA_1_8_HOME,JAVA_1_9_HOME</_removeheaders>
+            <_removeheaders.commons-parent>JAVA_1_3_HOME,JAVA_1_4_HOME,JAVA_1_5_HOME,JAVA_1_6_HOME,JAVA_1_7_HOME,JAVA_1_8_HOME,JAVA_1_9_HOME</_removeheaders.commons-parent>
```

Comment:
I don't understand what _removeheaders.commons-parent is doing?
Why is it necessary to change that line?

Reply:
See the explanations in https://github.com/bndtools/bnd/issues/4221#issuecomment-656258021 and https://github.com/apache/commons-lang/pull/578#issuecomment-656272161: `<_removeheaders>` overrides the defaults from `maven-bundle-plugin`, thus removing some values from this list that should have stayed there, in particular `Include-Resource`. On the other hand, using the *merged property* `<_removeheaders.commons-parent>` appends to the defaults, adding the values listed here to the ones already present in `maven-bundle-plugin`.

Reply:
I see.

This is not at all obvious, so it needs to be documented in the pom source with a pointer to the official documentation for the property syntax

Reply:
Done, the revised version of the PR has a description that should hopefully be more clear.

Reply:
Thanks, I've read the page in question, and it mentions merged properties but does not describe them as far as I can tell.

In particular it does not describe what the .commons-parent suffix represents.
It looks as though it may be related to the containing pom.
Is it the artifactId?
If so, why is ${project.artifactId} not used?
If not, how is the value chosen?

Reply:
The suffix is a user defined descriminator. The value is entirely informational but should be something unique-enough, that conveys some meaning for the use case. In this case it's something like _these are the `-removeheaders` properties set by the `commons-parent` pom_.

If you think the docs page can better describe this I can take a shot at making it more clear.

Reply:
Thanks.
The docs are not clear on how Bind merges properties.
For example, if a child pom uses the same discriminator as a parent pom, are the values merged or replaced, and if replaced, which takes precedence? This will obviously affect the choice of suffix.
It would be useful if the documentation provided some examples.

Reply:
Would you all and the maven-bundle-plugin folks mind participating in the thread "Reproducible builds, jars, bundles" on dev@maven.apache.org ? TY.